### PR TITLE
Yi kai

### PR DESCRIPTION
**Pull request by yikai03**
Added resumeRanking.py as model origin. Changes made in server.py. Provided a unit testing for resume ranking.py

### CODE REVIEW

File: backend/server.py [hunk 1]
```diff
@@ -221,18 +222,39 @@ async def readJob(job_id: str):
...
-        
-        
-    
+    # with open(f"resume/resume_in/{resume.filename}", "wb") as buffer:
+    #     shutil.copyfileobj(resume.file, buffer)#Test if needed, (should not be needed)
+    #Read job data
+    jobDesctiptionArray = []
+    jobTitleArray = []
+
+    for category in data:
```
Comment (SHINE-six) [inline backend/server.py]: need make amendment to fit with the new data structure

File: backend/server.py [hunk 2]
```diff
@@ -221,18 +222,39 @@ async def readJob(job_id: str):
...
+        return {"status": 200, "message": "Resume uploaded successfully"}
+    else:
+        return {"status": 400, "message": "Error uploading resume"}
 
 @app.get("/resume-ranking")
 async def getResumeRanking():
+    # Once HR select to view resume ranking
+    # Get resume from MongoDB
+    # Get job title to pull relative job description
     filePath = "resume/resume_ranking.json"
```
Comment (SHINE-six) [inline backend/server.py:292]: bring up during meeting

File: backend/server.py [hunk 3]
```diff
@@ -233,18 +234,39 @@ async def readJob(job_id: str):
```
Comment (SHINE-six) [inline backend/server.py:252]: def resumeParseBinary(resume):

File: backend/server.py (hunk 1 above)
Comment (SHINE-six) [reply backend/server.py]: uploadResume(resume, jobDetail :{jobposition, jobDesc, jobSkill}, email, uniqueResumeID)

toStoreJson = {
  "filename": "",
  "email": email,
  "uniqueResumeId": uniqueResumeID,
  "jobPostitionApply": jobPostitionApply,
  "AiDetection": "",
  "plagiarism": "",
  "suitability": "",
  "stage": "Ai detection", // suitability
  "pdfData": {}
}
parsedBinaryPDF = resumeParseBinary(resume):
toStoreJson['filename'] = parsedBinaryPDF['filename']
toStoreJson['pdfData'] = parsedBinaryPDF['pdfData']

//missing model
parsedPdfToText

toStoreJson['suitability'] = resumeSuitability.main()
toStoreJson['AiDetection'] = AiDetection.main(parsedPdfToText)['ai_similarity']
toStoreJson['plagiarism'] = plagiarism.main(parsedPdfToText)['resuklt']

Comment (SHINE-six) [reply backend/server.py]: print(mongoDB.postData("resumeData", toStoreJson))

**Review by SHINE-six — CHANGES_REQUESTED** [review]

File: backend/server.py (hunk 3 above)
Comment (yikai03) [reply backend/server.py:252]: Done

**Review by SHINE-six — APPROVED** [review]

**Pull request merged**